src/Symfony/Component/Security/Http/Firewall/X509AuthenticationListener.php

```diff
@@ -44,7 +44,10 @@ protected function getPreAuthenticatedData(Request $request)
         $user = null;
         if ($request->server->has($this->userKey)) {
             $user = $request->server->get($this->userKey);
-        } elseif ($request->server->has($this->credentialKey) && preg_match('#/emailAddress=(.+\@.+\..+)(/|$)#', $request->server->get($this->credentialKey), $matches)) {
+        } elseif ($request->server->has($this->credentialKey) && (
+                preg_match('#/emailAddress=(.+\@.+\..+)(/|$)#', $request->server->get($this->credentialKey), $matches)
+                || preg_match('/emailAddress=(.+\@.+\.[^,]*)/', $request->server->get($this->credentialKey), $matches)
```

Comment:
it'd be great to make these two calls into a single regexp match

Reply:
done